network/p2p/readConnection.go

```diff
@@ -67,31 +67,54 @@ func (rc *readConnection) receiveLoop(wg *sync.WaitGroup) {
 		}
 
 		var msg message.Message
-		// read the nex message (blocking call)
+		// read the next message (blocking call)
 		err := r.ReadMsg(&msg)
 
 		// error handling done similar to comm.go in pubsub (as suggested by libp2p folks)
 		if err != nil {
 			// if the sender closes the connection an EOF is received otherwise an actual error is received
 			if err != io.EOF {
 				rc.log.Error().Err(err)
-				err = rc.stream.Reset()
-				if err != nil {
-					rc.log.Error().Err(err)
-				}
+				rc.resetStream()
 			} else {
-				err = rc.stream.Close()
-				if err != nil {
-					rc.log.Error().Err(err)
-				}
+				rc.closeStream()
 			}
 			return
 		}
 
+		// check message size
+		maxSize := unicastMaxMsgSize(&msg)
+		if msg.Size() > maxSize {
+			// if message size exceeded, close stream and log error
+			rc.closeStream()
+			rc.log.Error().
+				Hex("sender", msg.OriginID).
+				Hex("event_id", msg.EventID).
+				Str("event_type", msg.Type).
+				Str("channel_id", msg.ChannelID).
+				Int("maxSize", maxSize).
+				Msg("received message exceeded permissible message maxSize")
+			return
+		}
+
 		// log metrics with the channel name as OneToOne
 		rc.metrics.NetworkMessageReceived(msg.Size(), metrics.ChannelOneToOne, msg.Type)
 
 		// call the callback
 		rc.callback(&msg)
 	}
 }
+
+func (rc *readConnection) closeStream() {
+	err := rc.stream.Close()
+	if err != nil {
+		rc.log.Error().Err(err)
+	}
+}
+
+func (rc *readConnection) resetStream() {
+	err := rc.stream.Reset()
+	if err != nil {
+		rc.log.Error().Err(err)
```

Comment:
Please add a message to the logs on these two methods (e.g., `could not close stream`). It makes it drastically easier to spot the problem.